### PR TITLE
Fix mapbase test warnings

### PR DESCRIPTION
**Pull request by dstansby**
Taken out of #3648

### CODE REVIEW

File: sunpy/map/tests/test_mapbase.py [hunk 1]
```diff
@@ -82,7 +95,10 @@ def generic_map():
 
 def test_fits_data_comparison(aia171_test_map):
     """Make sure the data is the same in pyfits and SunPy"""
-    data = fits.open(os.path.join(testpath, 'aia_171_level1.fits'))[0].data
+    with warnings.catch_warnings():
```
Comment (Cadair) [inline sunpy/map/tests/test_mapbase.py]: out of interest, why the `warnings.catch_warnings` here and not `pytest.warns`?

Comment (nabobalis) [reply sunpy/map/tests/test_mapbase.py]: We don't care about the warning?

Comment (dstansby) [reply sunpy/map/tests/test_mapbase.py]: Um I'm not sure, let me get back to you on that one...

Comment (dstansby) [reply sunpy/map/tests/test_mapbase.py]: Fixed

**Pull request merged**